### PR TITLE
Fix TypeScript Module Bundling

### PR DESCRIPTION
**Pull request by OR13**
- use peer dependencies
- prepare module for lerna inclusion (remove release scripts)
- add docs to root dir
- add tests for react and node of typescript module bundle

These changes should only be element-core module and element-module tests.

The idea is to include element-core as a dependency of element-lib, and slowly move element-lib to element core, adding type-safety and documentation as we go.

### CODE REVIEW

File: packages/element-modules/src/Node/ElementCore.Node.spec.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const ElementCore = require("@transmute/element-core");
```
Comment (gjgd) [inline packages/element-modules/src/Node/ElementCore.Node.spec.js]: Looks like `ElementCore.Node.spec.js` and `ElementCore.React.spec.js ` are the same.

I assume it is to test that element-core works both in the browser, and in a node environment but don't see how the environment could be different between the two files

File: packages/element-api/package.json [hunk 2]
```diff
@@ -20,6 +21,7 @@
   },
   "dependencies": {
     "@google-cloud/logging-winston": "^0.10.2",
+    "@transmute/element-core": "^0.0.0",
```
Comment (gjgd) [inline packages/element-api/package.json]: Is this used somewhere?

File: packages/element-app/package.json [hunk 3]
```diff
@@ -8,6 +8,7 @@
     "@material-ui/core": "^3.9.3",
     "@material-ui/icons": "^3.0.2",
     "@transmute/did-wallet": "0.0.0-3",
+    "@transmute/element-core": "^0.0.0",
```
Comment (gjgd) [inline packages/element-app/package.json]: Is this used somewhere?

File: packages/element-lib/package.json [hunk 4]
```diff
@@ -38,6 +38,7 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "@transmute/element-core": "^0.0.0",
```
Comment (gjgd) [inline packages/element-lib/package.json]: Is this used somewhere?

File: packages/element-api/package.json (hunk 2 above)
Comment (OR13) [reply packages/element-api/package.json]: no it is not, should be removed

File: packages/element-modules/src/Node/ElementCore.Node.spec.js (hunk 1 above)
Comment (OR13) [reply packages/element-modules/src/Node/ElementCore.Node.spec.js]: yep

File: packages/element-app/package.json (hunk 3 above)
Comment (OR13) [reply packages/element-app/package.json]: its an error, I am removing it.

File: packages/element-lib/package.json (hunk 4 above)
Comment (OR13) [reply packages/element-lib/package.json]: another error, from testing, I am removing it.

**Comment by OR13** [comment]
> Couldn't review the docs part because it's all html/css. Is it published somewhere?

It will be published when this gets merged : )

The idea is to pull docs into the root repo and publish them on master, via github-pages.

File: packages/element-modules/src/Node/ElementCore.Node.spec.js (hunk 1 above)
Comment (gjgd) [reply packages/element-modules/src/Node/ElementCore.Node.spec.js]: yep as in, "yes they are the same and one can be deleted" or "yes they run in different environment"?

In the latter case, how?

**Comment by gjgd** [comment]
Noted, then part of the review process should be to test this doc locally.

File: packages/element-modules/src/Node/ElementCore.Node.spec.js (hunk 1 above)
Comment (OR13) [reply packages/element-modules/src/Node/ElementCore.Node.spec.js]: https://github.com/decentralized-identity/element/pull/45/files#diff-b9800c3aab96bca06c080a5c3f971140R14

using jest `--env` to set the environment and run each file.

**Comment by OR13** [comment]
> Noted, then part of the review process should be to test this doc locally.

Yes, I should update the readme.

I use `npm i -g http-server` and `serve ./docs`

File: packages/element-modules/src/Node/ElementCore.Node.spec.js (hunk 1 above)
Comment (gjgd) [reply packages/element-modules/src/Node/ElementCore.Node.spec.js]: Ah yes, didn't see that mb.

**Pull request merged**